### ISSUE TITLE
skip items with nil values

### ISSUE THREAD

**Issue by skcc321** [procore/blueprinter#124]
Is it possible to have an option for ignoring nil values like this one in JBuilder https://www.rubydoc.info/github/rails/jbuilder/Jbuilder:ignore_nil!
I'm using JBuilder everywhere, and we have a convention to skip items with nil values (client wants it). I'm going to use Blueprinter instead, but the last thing which blocks to do that is this missing option.

**Comment by mcclayton** [comment]
Hi @skcc321, thanks for reaching out. 
I'll bring this up to the team and see if (and in what way if so) we'd like to support something like this.
In the meantime, this is something that can currently be accomplished using Blueprinter's [conditional field functionality](https://github.com/procore/blueprinter#conditional-field).
i.e.: 

```
field :name, unless: ->(user, _options) { user.name.nil? }
```

Though obviously this is not ideal to have to specify for every single one of your Blueprint fields. I'll re-post when I get the team's feedback on this.

**Comment by anlek** [comment]
I have had this issue with `date_format: "%d %m %Y"` and the date being `nil`.  It would be great if the result would be `null` instead of an error about not being able to convert `nil` to a date object.

**Comment by mcclayton** [comment]
Hi @anlek, can you please verify for me what version of Blueprinter you are using? That specific issue you are seeing for nil dates ought to be fixed in versions >= `0.8.0`.
https://github.com/procore/blueprinter/pull/117
https://github.com/procore/blueprinter/blob/master/CHANGELOG.md#080----20181119

**Comment by mcclayton** [comment]
@skcc321 
Sorry for the delay in getting back to you, I brought it up to the team and we are considering a more general solution that would hopefully suffice your issue.
We are considering implementing global `:if` and `:unless` options in the Blueprinter config.
i.e.
```ruby
config.if = -> { obj.is_a?(Foo) }
```

```ruby
class SomeBluepint
  field :stuff, if: -> { obj.is_a?(Bar) }
end
```
Where field-level configurations will override the global configuration, but the global configuration would be used if no field-level option was provided. 
I think implementing this would solve your issue if I am understanding you correctly?

**Comment by anlek** [comment]
@mcclayton You're right, I was using `0.6.0` version. I'll update and look forward to the fix. As for the global if/unless. It's an interesting idea, I think it's a good place to build a catch all for a given application. Thanks for the details!

**Comment by skcc321** [comment]
Thank you @mcclayton for taking care of this issue. I appreciate it a lot.
The suggested approach gives the ability to achieve expected behaviour.
But, it would be great to be able to set skip_nil per Blueprint serializer, but not global.
For example, I have external integration with third-party service which wants to skip_nils.
but also I have another integration wich wants nils.

the described case can be covered easily with JBuilder directly inside view https://www.rubydoc.info/github/rails/jbuilder/Jbuilder:ignore_nil!

Is there any problem to put option on some Plueprint serializer (parent serializer) so all nested serializer will behave the same way?

Anyway, I'm excited about the current solution. So thank you for what you have done.

**Comment by mcclayton** [comment]
@skcc321 Ahh that makes sense. I've been sick recently and been wrapped up in some other work -- but once I feel better and wrap up my other work, I'll put some thought into an inheritable configuration  solution 👍

**Comment by mcclayton** [comment]
I'm going to go ahead and close this, as this should now be possible using global if/unless blocks as of https://github.com/procore/blueprinter/pull/131 being merged and with the `0.18.0` release.
